packages/http/src/validator/__tests__/functional.spec.ts

```diff
@@ -31,6 +31,87 @@ describe('HttpValidator', () => {
       });
     });
 
+    describe('deprecated keyword validation', () => {
+      const resource = {
+        id: "abc",
+        method: "get",
+        path: "/test",
+        responses: [
+          {
+            code: "200",
+            contents: [
+              {
+                mediaType: "application/json",
+                schema: {
+                  type: "object",
+                  properties: {
+                    prop: {
+                      type: "string",
+                      example: "abc"
+                    }
+                  },
+                  $schema: "http://json-schema.org/draft-04/schema#"
+                },
+              }
+            ]
+          }
+        ],
+        request: {
+          query: [
+            {
+              style: HttpParamStyles.Form,
+              deprecated: true,
+              name: "productId"
+            }
+          ],
+        },
+      };
+
+      it('returns warnings', () => {
+        expect(
+          validateInput({
+            // @ts-ignore
```

Comment:
Why is this required?

Reply:
@XVincentX, this is a mystery:
![a_mystery](https://user-images.githubusercontent.com/24682079/67269259-ad495400-f4b6-11e9-9f5f-477246822f54.gif)

Reply:
Leave it there, I'll figure it out and push a fix.